invokeai/app/services/events/events_base.py

```diff
@@ -404,53 +404,72 @@ def emit_download_error(self, source: str, error_type: str, error: str) -> None:
             },
         )
 
-    def emit_model_install_started(self, source: str) -> None:
+    def emit_model_install_downloading(
+        self,
+        source: str,
+        local_path: str,
+        bytes: int,
+        total_bytes: int,
+        parts: List[Dict[str, Any]],
```

Comment:
Has been narrowed. Can I go ahead and pass pydantic objects to `parts` now? The reason I passed a dict is only because I saw simple objects being used elsewhere in the events module and figured it was a limitation of the events library.

Reply:
Oh right. Yeah I don't think it's going to accept pydantic objects without special handling elsewhere. Maybe we should just leave it as-is then.

Reply:
The latest version of `fastapi-events` now supports directly dispatching pydantic models. You can model event payloads as pydantic models, with an extra private event name attr, then dispatch it directly. Pretty slick. Anyways, future enhancement.

Reply:
For the future...